framework/src/utils/PerfNode.C

```diff
@@ -0,0 +1,48 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PerfNode.h"
+
+PerfNode::PerfNode(unsigned int id) : _id(id), _total_time(0) {}
+
+PerfNode *
+PerfNode::getChild(unsigned int id)
+{
+  auto & child_node = _children[id];
+
+  if (!child_node)
+    child_node.reset(new PerfNode(id));
```

Comment:
libmesh_make_unique

Reply:
👍 Honestly didn't know the right thing to do to reinitialize a `unique_ptr`...

Reply:
@permcody Actually - as far as I can tell you can't use `libmesh_make_unique` to pass in to `reset()`.  Any better ideas?